util/grpcerrors/grpcerrors.go

```diff
@@ -45,7 +46,7 @@ func ToGRPC(ctx context.Context, err error) error {
 
 	// If the original error was wrapped with more context than the GRPCStatus error,
 	// copy the original message to the GRPCStatus error
-	if err.Error() != st.Message() {
+	if errorHasMoreContext(err, st) {
```

Comment:
What is the case where we don't want the original details (or ones set in FromGRPC) to not be added into this error?

Reply:
`AsGRPCStatus()` unwraps until it finds something implementing `GRPCStatus()`. So if we had something like `errors.Wrap(errors.Wrap(st.Err(), "foo") "bar")`, it would discard the `foo: bar:` prefix otherwise.